src/chain/mod.rs

```diff
@@ -54,15 +64,16 @@ pub async fn start_chain(port: u16) -> Result<Child> {
 
 /// kit chain, alias to anvil
 pub async fn execute(port: u16) -> Result<()> {
-    fetch_kinostate().await?;
+    let state_hash = write_kinostate().await?;
+    let state_path = format!("./kinostate-{}.json", state_hash);
```

Comment:
I don't like that we do this same thing in 3 places. Seems like it'll be easy to break in future when we accidentally update only m<n of n items. Not sure how to do better tho.

Reply:
that's fair, given that they're always used together right now, I can put write_kinostate() into start_chain()